test-app/tests/integration/components/bs-dropdown/button-test.js

```diff
@@ -16,4 +16,43 @@ module('Integration | Component | bs-dropdown/button', function (hooks) {
     assert.dom('button').exists('dropdown button is a button');
     assert.dom('.dropdown-toggle').exists('has dropdown-toggle class');
   });
+
+  test('dropdown button has coorect css type class', async function (assert) {
+    await render(
+      hbs`<BsDropdown as |dd|><dd.button @type="primary">Test</dd.button></BsDropdown>`,
+    );
+
+    assert
+      .dom('button')
+      .hasClass('btn-primary', 'dropdown button has primary type');
+  });
+
+  test('dropdown button has correct css outline class', async function (assert) {
+    await render(
+      hbs`<BsDropdown as |dd|><dd.button @outline={{true}}>Test</dd.button></BsDropdown>`,
+    );
+
+    assert
+      .dom('button')
+      .hasClass(
+        'btn-outline-secondary',
+        'dropdown button has secondary outline type',
+      );
+  });
+
+  test('dropdown button has correct size', async function (assert) {
+    await render(
+      hbs`<BsDropdown as |dd|><dd.button @size="lg">Test</dd.button></BsDropdown>`,
```

Comment:
Prettify the inline template:

```suggestion
      hbs`<BsDropdown as |dd|><dd.button @size='lg'>Test</dd.button></BsDropdown>`,
```